app/views/pages/search.html.erb

```diff
@@ -161,4 +180,39 @@
             <% end %>
         </div>
     </div>
-</div>
\ No newline at end of file
+</div>
+<%#= google map %>
+    <div>
+        <div id="map-search" class="mt-3 mb-3"></div>
+        <script src="https://maps.googleapis.com/maps/api/js?key=AIzaSyCfXoNJFnAa8MmCNVSxj2x1iFUmRVXk_xc"></script>
+
+        <script>
+            function initialize() {
+                var location = {lat: 44.5403, lng: -78.5463};
+                <% if @rooms_array.length > 0 %>
+                    location = {lat: Number("<%= @rooms_array[0].latitude %>"), lng: Number("<%= @rooms_array[0].longitude %>")};
+                <% end %>
+                var map = new google.maps.Map(document.getElementById('map-search'),{
+                    center: location,
+                    zoom: 14
+                });
+
+                var marker, infoWindow;
+                <% @rooms_array.each do |room|%>
+                    marker = new google.maps.Marker({
+                        position: {lat: Number("<%= room.latitude %>"), lng: Number("<%= room.longitude %>")},
+                        map: map
+                    });
+
+                    infoWindow = new google.maps.InfoWindow({
+                        content: '<div id="content"><%= image_tag room.photos.first.image, class: "w-100" %></div>'
+                    });
+                <% end %>
+
+                infoWindow.open(map, marker);
+            }
+
+            google.maps.event.addDomListener(window,'load', initialize);
+        </script>
+    </div>
+    <%#= /google map %>
```

Comment:
Remove extra indention for the google map code.
You have to move it to the left.